plugin/kubernetes/controller.go

```diff
@@ -125,7 +125,7 @@ func newdnsController(ctx context.Context, kubeClient kubernetes.Interface, opts
 			&api.Pod{},
 			cache.ResourceEventHandlerFuncs{AddFunc: dns.Add, UpdateFunc: dns.Update, DeleteFunc: dns.Delete},
 			cache.Indexers{podIPIndex: podIPIndexFunc},
-			object.DefaultProcessor(object.ToPod(opts.skipAPIObjectsCleanup)),
+			object.DefaultProcessor(object.ToPod(opts.skipAPIObjectsCleanup), nil),
```

Comment:
this boolean arg is is only used for a single test? that seems excessive for adding it?

Reply:
The boolean arg was already there. IIUC, it was added when the record latency metrics were added, to allow the test to work.

The new parameter is a function that calculates the record latency metrics for that object type.  Currently it's only implemented for the Endpoints object.  So, for Services and Pods, we pass nil.